### ISSUE TITLE
Files with invalid mtime cause error

### ISSUE THREAD

**Issue by alphaniner** [fman-community/fman#668]
When entering a folder with files that have invalid modification time timestamps, fman throws the following error:
```
Column 'Modified' raised error.

Traceback (most recent call last):
  File "fman\impl\plugins\plugin.py", line 389, in get_str
  File "C:\Users\alphaniner\AppData\Local\fman\Versions\1.6.4\Plugins\Core\core\__init__.py", line 99, in get_str
    mtime_qt = QDateTime.fromMSecsSinceEpoch(int(mtime.timestamp() * 1000))
OSError: [Errno 22] Invalid argument
```
If the error is dismissed it just comes up again, indefinitely. Following your example I just used a `try/except OSError` to `return ''` in such cases.

**Comment by mherrmann** [comment]
What kind of files / folder does this occur for? Is it a local folder? A network drive? A custom file system? Which operating system are you on?

**Comment by alphaniner** [comment]
I only had the issue in one directory before applying the fix I mentioned. The affected files were downloaded archives of various types (7z, zip, rar). I've since fixed the timestamps, so unless I find other files with the same issue I won't be able to reproduce.

The directory is on an NTFS filesystem on a local SATA HDD. Most of the files had originally been downloaded to an ImDisk Ramdisk and moved or copied (with Windows File Explorer) to the directory. 

However, only a few files had invalid timestamps so none of these things jump out as a common denominator.

OS is Windows 7.

**Comment by mherrmann** [comment]
Interesting, thank you.

> Most of the files had originally been downloaded to an ImDisk Ramdisk and moved or copied (with Windows File Explorer) to the directory.

I suspect that the move from Ramdisk to your local HDD caused the invalid timestamps.

What I don't understand is where an `OSError` can occur in the line

    QDateTime.fromMSecsSinceEpoch(int(mtime.timestamp() * 1000))

It consists of three calls:

    mtime.timestamp()
    int(mtime.timestamp() * ...)
    QDateTime.fromMSecsSinceEpoch(...)

The first is from Python's `datetime` class. This should not throw an `OSError`. The `int(...)` call also should not raise such an error. (I'd be happy for it to raise a `ValueError`.) And the `QDateTime` call also does not raise an `OSError` (I tried). So what gives? Do you have an idea / experience from when you were fixing the problem?

**Comment by alphaniner** [comment]
I found another file with invalid mtime, and broke the line down as you indicated. It's `mtime.timestamp()` that throws the error.

Windows File Explorer also fails to show anything in the 'Modification Time' field, but the Properties dialog shows "Modified: Wednesday, ‎December ‎31, ‎1969, ‏‎7:00:00 PM". According to [this](https://superuser.com/questions/796762), that actually is a valid timestamp on NTFS despite it not showing in Explorer.

More to the point, I noticed it predates the common epoch.

Putting everything together I eventually found [this](https://stackoverflow.com/questions/42413138).
